controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -8,8 +16,89 @@
 
 #include <easylogging++.h>
 
+#include <algorithm>
+
 using namespace beerocks;
 
+std::string wfa_ca::wfa_ca_status_to_string(eWfaCaStatus status)
+{
+    switch (status) {
+    case eWfaCaStatus::RUNNING:
+        return "RUNNING";
+    case eWfaCaStatus::INVALID:
+        return "INVALID";
+    case eWfaCaStatus::ERROR:
+        return "ERROR";
+    case eWfaCaStatus::COMPLETE:
+        return "COMPLETE";
+    default:
+        LOG(ERROR) << "Status argument doesn't have an enum";
+        break;
+    }
+    return std::string();
+}
+
+wfa_ca::eWfaCaCommand wfa_ca::wfa_ca_command_from_string(std::string command)
+{
+    std::transform(command.begin(), command.end(), command.begin(), ::toupper);
+
+    if (command == "CA_GET_VERSION") {
+        return CA_GET_VERSION;
+    } else if (command == "DEV_CONFIGURE_IE") {
+        return DEV_CONFIGURE_IE;
+    } else if (command == "DEV_GET_FRAME_INFO") {
+        return DEV_GET_FRAME_INFO;
+    } else if (command == "DEVICE_GET_INFO") {
+        return DEVICE_GET_INFO;
+    } else if (command == "DEV_EXEC_ACTION") {
+        return DEV_EXEC_ACTION;
+    } else if (command == "DEV_RESET_DEFAULT") {
+        return DEV_RESET_DEFAULT;
+    } else if (command == "DEV_SEND_1905") {
+        return DEV_SEND_1905;
+    } else if (command == "DEV_SEND_FRAME") {
+        return DEV_SEND_FRAME;
+    } else if (command == "DEV_SET_CONFIG") {
+        return DEV_SET_CONFIG;
+    }
+
+    return WFA_CA_COMMAND_MAX;
+}
+
+void wfa_ca::reply_invalid_error(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx,
+                                 eWfaCaStatus status, const std::string &description)
+{
+    if (status != eWfaCaStatus::INVALID || status != eWfaCaStatus::ERROR) {
+        LOG(WARNING) << "function misuse, status is not INVALID or ERROR";
+        return;
+    }
+
+    std::string reply_string =
+        "status," + wfa_ca_status_to_string(status) + ",errorCode," + description;
+    reply(sd, cmdu_tx, reply_string, false);
+}
+
+bool wfa_ca::reply(Socket *sd, ieee1905_1::CmduMessageTx &cmdu_tx, const std::string &reply_string,
+                   bool pending_second_response)
```

Comment:
Instead of calling it `pending_second_response`, call it `pending_additional_response` like you called it below.